packages/expo-analytics-segment/android/src/main/java/expo/modules/analytics/segment/SegmentModule.kt

```diff
@@ -0,0 +1,222 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package expo.modules.analytics.segment
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.segment.analytics.Analytics
+import com.segment.analytics.Options
+import com.segment.analytics.Properties
+import com.segment.analytics.Traits
+import com.segment.analytics.android.integrations.firebase.FirebaseIntegration
+import expo.modules.interfaces.constants.ConstantsInterface
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ExpoMethod
+import java.util.*
+
+class SegmentModule(private val mContext: Context) : ExportedModule(mContext) {
+  private var mClient: Analytics? = null
+  private var mConstants: ConstantsInterface? = null
+
+  // We have to keep track of `enabled` on our own.
+  // Since we have to change tag every time (see commit 083f051), Segment may or may not properly apply
+  // remembered preference to the instance. The module in a standalone app would start disabled
+  // (settings = { 0 => disabled }, tag = 0) but after OTA update it would reload with
+  // (settings = { 0 => disabled }, tag = 1) and segment would become enabled if the app does not
+  // disable it on start.
+  private val mSharedPreferences: SharedPreferences = mContext.getSharedPreferences(NAME, Context.MODE_PRIVATE)
+
+  override fun getName(): String {
+    return NAME
+  }
+
+  @ExpoMethod
+  fun initialize(writeKey: String, promise: Promise) {
+    val builder = Analytics.Builder(mContext, writeKey)
+    builder.experimentalUseNewLifecycleMethods(false)
+    builder.tag((sCurrentTag++).toString())
+    builder.use(FirebaseIntegration.FACTORY)
```

Comment:
```suggestion
    val builder = Analytics.Builder(mContext, writeKey)
   	  .experimentalUseNewLifecycleMethods(false)
      .tag((sCurrentTag++).toString())
      .use(FirebaseIntegration.FACTORY)
```
This should be possible 😉